packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -49,41 +45,84 @@ function NavigationMenuItemEdit( {
 	isParentOfSelectedBlock,
 	setAttributes,
 	insertMenuItemBlock,
+	fetchSearchSuggestions,
 } ) {
+	const { label, opensInNewTab, title, url } = attributes;
+	const link = title ? { title, url } : null;
 	const [ isLinkOpen, setIsLinkOpen ] = useState( false );
-	const [ isEditingLink, setIsEditingLink ] = useState( false );
-	const [ urlInput, setUrlInput ] = useState( null );
+	const [ wasClosedByLinkControl, setWasClosedByLinkControl ] = useState( false );
+
+	/**
+	 * It's a kind of hack to handle closing the LinkControl popover
+	 * clicking on the ToolbarButton link.
+	 */
+	useEffect( () => {
+		if ( ! isSelected ) {
+			setIsLinkOpen( false );
+			setWasClosedByLinkControl( false );
+		}
+		return () => {
+			setIsLinkOpen( false );
+			setWasClosedByLinkControl( false );
+		};
+	}, [ isSelected ] );
+
+	// Set the menu item when it's new.
+	// - Open LinkControl popover.
+	useEffect( () => {
+		if ( ! label && isSelected ) setIsLinkOpen( true );
```

Comment:
Instead of initializing the state of is Link open to false could we initialize it to `(! label && isSelected)`, avoiding the need for this effect and avoid an extra rerender?